### PR TITLE
[Ops][Feature] Add custom op triton_split_qkv_rmsnorm_mrope

### PR DESCRIPTION
**Pull request by fuzhihong699**
### What this PR does / why we need it?

This PR adds a new custom Triton operator `triton_split_qkv_rmsnorm_mrope`. This operator fuses three operations:
- Splitting the input tensor into Q, K, and V tensors.
- Applying RMS Normalization to Q and K tensors.
- Applying Mixture of Rotary Position Embeddings (mRoPE) to Q and K tensors.

This fused kernel is expected to improve performance by reducing kernel launch overhead and memory access, , primarily aimed at optimizing Qwen3-Omni-30B.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

- vLLM version: v0.15.0
- vLLM main: https://github.com/vllm-project/vllm/commit/v0.15.0

### How to used?

edit vllm/vllm/model_executor/models/qwen3_moe.py, referring to the following:

```
class Qwen3MoeAttention(nn.Module):
	...

    def forward(
        self,
        positions: torch.Tensor,
        hidden_states: torch.Tensor,
    ) -> torch.Tensor:
        qkv, _ = self.qkv_proj(hidden_states)

        # q, k, v = qkv.split([self.q_size, self.kv_size, self.kv_size], dim=-1)
        # # Add qk-norm
        # q_by_head = q.view(*q.shape[:-1], q.shape[-1] // self.head_dim, self.head_dim)
        # q_by_head = self.q_norm(q_by_head)
        # q = q_by_head.view(q.shape)

        # k_by_head = k.view(*k.shape[:-1], k.shape[-1] // self.head_dim, self.head_dim)
        # k_by_head = self.k_norm(k_by_head)
        # k = k_by_head.view(k.shape)
        # q, k = self.rotary_emb(positions, q, k)

        cos_sin = self.rotary_emb.cos_sin_cache[positions]
        q, k, v = torch.ops.vllm.triton_split_qkv_rmsnorm_mrope(
            qkv=qkv,
            q_weight=self.q_norm.weight,
            k_weight=self.k_norm.weight,
            cos_sin=cos_sin,
            num_q_heads=self.num_heads,
            num_kv_heads=self.num_kv_heads,
            head_size=self.head_dim,
            eps=self.q_norm.variance_epsilon,
            mrope_section=self.rotary_emb.mrope_section,
            q_bias=self.q_norm.bias,
            k_bias=self.k_norm.bias,
        )

        attn_output = self.attn(q, k, v)
        output, _ = self.o_proj(attn_output)
        return output
```

Like this commit:
https://github.com/vllm-project/vllm/commit/3a8756298a1f3ddab1ffff1852ada3f2ac3dde35

### Result

Test results on the Qwen3-Omni-30B model, 300 concurrent requests.

|            | Concurrency | TTFT(ms) | TPOT(ms) | Output Token Throughput(token/s) |
| ---------- | ----------- | -------- | -------- | -------------------------------- |
| **before** | 300         | 13799.97 | 62.96    | 3301.87                          |
| **after**  | 300         | 11931.85 | 54.93    | 3854.65                          |

### CODE REVIEW

**Comment by wangxiyuan** [comment]
how/where this ops will be used?

**Comment by fuzhihong699** [comment]
> how/where this ops will be used?

edit vllm/vllm/model_executor/models/qwen3_moe.py, referring to the following:

class Qwen3MoeAttention(nn.Module):
	...

    def forward(
        self,
        positions: torch.Tensor,
        hidden_states: torch.Tensor,
    ) -> torch.Tensor:
        qkv, _ = self.qkv_proj(hidden_states)

        # q, k, v = qkv.split([self.q_size, self.kv_size, self.kv_size], dim=-1)
        # # Add qk-norm
        # q_by_head = q.view(*q.shape[:-1], q.shape[-1] // self.head_dim, self.head_dim)
        # q_by_head = self.q_norm(q_by_head)
        # q = q_by_head.view(q.shape)

        # k_by_head = k.view(*k.shape[:-1], k.shape[-1] // self.head_dim, self.head_dim)
        # k_by_head = self.k_norm(k_by_head)
        # k = k_by_head.view(k.shape)
        # q, k = self.rotary_emb(positions, q, k)

        cos_sin = self.rotary_emb.cos_sin_cache[positions]
        q, k, v = torch.ops.vllm.triton_split_qkv_rmsnorm_mrope(
            qkv=qkv,
            q_weight=self.q_norm.weight,
            k_weight=self.k_norm.weight,
            cos_sin=cos_sin,
            num_q_heads=self.num_heads,
            num_kv_heads=self.num_kv_heads,
            head_size=self.head_dim,
            eps=self.q_norm.variance_epsilon,
            mrope_section=self.rotary_emb.mrope_section,
            q_bias=self.q_norm.bias,
            k_bias=self.k_norm.bias,
        )

        attn_output = self.attn(q, k, v)
        output, _ = self.o_proj(attn_output)
        return output

**Comment by fuzhihong699** [comment]
refer: https://github.com/vllm-project/vllm-ascend/pull/6730
